cluster-provision/k8s/1.21/istio.sh

```diff
@@ -31,17 +32,33 @@ spec:
       excludeNamespaces:
        - istio-system
        - kube-system
-      logLevel: debug 
+      logLevel: debug
     sidecarInjectorWebhook:
       injectedAnnotations:
         "k8s.v1.cni.cncf.io/networks": istio-cni
 EOF
 
-retries=0
-while [[ $retries -lt 20 ]]; do
-  echo "waiting for istio-cni-node daemonset"
-  sleep 5
-  kubectl --kubeconfig /etc/kubernetes/admin.conf get daemonset istio-cni-node -n kube-system && break
-  retries=$((retries + 1))
-done
-
+cat <<EOF >/opt/istio/istio-operator.cr.yaml
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  namespace: istio-system
+  name: istio-operator
+spec:
+  profile: demo
+  components:
+    cni:
+      enabled: true
+      namespace: kube-system
+  values:
+    global:
+      jwtPolicy: first-party-jwt
+    cni:
+      chained: true
+      cniBinDir: /opt/cni/bin
+      cniConfDir: /etc/cni/net.d
+      excludeNamespaces:
+       - istio-system
+       - kube-system
+      logLevel: debug
+EOF
```

Comment:
What about patching the current one with 
```
sidecarInjectorWebhook:
      injectedAnnotations:
        "k8s.v1.cni.cncf.io/networks": istio-cni
```
instead of creating a new CR ?

Reply:
There are a few more differences
`      cniConfDir: /etc/cni/net.d` vs `      cniConfDir: /etc/cni/multus/net.d`
and 
`      chained: true` vs `      chained: false`

But yeah I thought about this. The thing is patching needs to be done in `cluster-up/cluster/k8s-provider-common.sh`, because I don't have access to `KUBEVIRT_WITH_CNAO` env var in `istio.sh`. I decided to keep the logic of `k8s-provider-common.sh` simple.

Reply:
I mean creating a pair of yamls but reduce duplication by patching and we can do the following

For `cniConfDir` and `chained` we can use envsubst

For the `sidecarInjectorWebhook` we can append at the end.